src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -85,4 +131,338 @@ public void requestMessageFromIris(IrisSession session) {
         // TODO: Future: Switch between different session types
         irisChatService.requestAndHandleResponse(session);
     }
+
+    /**
+     * Creates the initial system message for the session and set LLM
+     *
+     * @param session The session to generate the initial system message for
+     * @return The created IrisMessage for storing in the database
+     */
+    public IrisMessage createInitialSystemMessage(IrisSession session) {
+        var exercise = session.getExercise();
+        String title = "undefined";
+        if (exercise.getCourseViaExerciseGroupOrCourseMember() != null && exercise.getCourseViaExerciseGroupOrCourseMember().getTitle() != null) {
+            title = exercise.getCourseViaExerciseGroupOrCourseMember().getTitle();
+        }
+        String initialSystemMessage = INITIAL_PROMPT_TEMPLATE.formatted(title, ProblemStatementUtils.truncateProblemStatement(exercise.getProblemStatement()));
+
+        var messageContent = new IrisMessageContent();
+        messageContent.setTextContent(initialSystemMessage);
+        var message = new IrisMessage();
+        message.setContent(List.of(messageContent));
+        message.setSender(IrisMessageSender.ARTEMIS);
+        message.setSentAt(ZonedDateTime.now());
+        message.setSession(session);
+        return message;
+    }
+
+    /**
+     * Utility class for removing all unnecessary information from problem statements
+     */
+    private static class ProblemStatementUtils {
+
+        /**
+         * Removes all unnecessary information from problem statements
```

Comment:
I think it's important to mention here, that it not only removes unnecessary information but also returns only the first 2000 tokens. Especially for really long problem statements this is not even the first 20% of the task. For now it's okay, but we should also add a TODO to provide a better way of summarizing the problem statement in the future